apps/communications/dialer/test/marionette/lib/views/keypad/views.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+/* global module */
+var KeypadAccessor = require('./accessors');
+
+function KeypadView(client) {
+  
+  this.client = client;
+  this.accessors = new KeypadAccessor(client);
+
+}
+
+KeypadView.prototype = {
+ 
+  key: function(keyNumber) {
+    return this.client.helper.waitForElement
+      (this._getKeypadKeySelector(keyNumber));
+  },
+  _getKeypadKeySelector: function(key) {
+    return '.keypad-key[data-value="' + key +'"]';
+  },
+  
+  typePhoneNumber: function(number) {
+    var i;
+    var typeNumberArray = String(number).split('');
+
+    for (i in typeNumberArray){
+      if(i == '+'){
+        var zero_key = parseInt(typeNumberArray[i]);
+        var zero = this.key[zero_key];
+        this.actions.longPress(zero, 1).perform();
+      }
+      else{
+        var key_tapped = parseInt(typeNumberArray[i]);
+        var other_key = this.key(key_tapped);
```

Comment:
That variable doesn't add much value to the code. It's also a bit ambiguous. We can simplify it with just 1 line:
this.key(i).tap();